### ISSUE TITLE
Saving elements without urls expires entire cache.

### ISSUE THREAD

**Issue by jodeba** [putyourlightson/craft-blitz#598]
Hi, I've encountered a problem after upgrading Blitz

## Steps to reproduce the issue.
- mode `2`: Expire the cache and regenerate in a queue job
- create/save a User, it does not have a url
- there is a queue job:
```
{
    "description": null,
    "data": {
        "cacheIds": [],
        "elements": {
            "craft\\elements\\User": {
                "elementIds": {
                    "954644": true
                },
                "changedAttributes": {
                    "954644": {
                        // all attributes
                    }
                },
                "changedFields": {
                    "954644": {
                       // some fields
                    }
                },
                "isChangedByAttributes": {
                    "954644": false
                },
                "isChangedByFields": {
                    "954644": false
                }
            }
        }
    },
    "forceClear": false,
    "forceGenerate": false,
    " craft\\queue\\BaseJob _progress": 0,
    " craft\\queue\\BaseJob _progressLabel": null
}
```
- look at table 'blitz_caches': expiryDate is set for all records.

## Problem
I think it is because [this function](https://github.com/putyourlightson/craft-blitz/blob/c915a644f0435f7709a61f4b7d2c75eefea49316/src/helpers/SiteUriHelper.php#L324) returns all cacheIds for an empty array of siteUris

## Blitz version
Started after upgrading to Blitz 4.9.3, issue not present in Blitz 4.7.1. Running on Craft 4.5.12

**Comment by bencroker** [comment]
Thanks for reporting this! Fixed in https://github.com/putyourlightson/craft-blitz/commit/714cc8a6cdd6299f806d500dfa9f30c249c4fcaa for the next release.

You can test this by running `composer require "putyourlightson/craft-blitz:dev-develop as 4.9.5"`.

**Comment by jodeba** [comment]
Thank you!

**Comment by bencroker** [comment]
Released in 4.10.0.
